Content.Shared/Verbs/VerbCategory.cs

```diff
@@ -0,0 +1,86 @@
+        /// </remarks>
+        public readonly bool Contractible;
+
+        /// <summary>
+        ///     Maximum verb text length allowed when contracting.
+        /// </summary>
+        /// <remarks>
+        ///     <par> Sometimes contracting verb categories can result in extremely long verbs. For example: "Insert
+        ///     Valerie Schaeffer's ID card (assistant)". In order to avoid extremely long verb text, and consequently,
+        ///     large context menu popups, this limits the length when contracting. </par>
+        ///
+        ///     <para>With this option, the above example would not be contracted, but something like "Eject pen" could
+        ///     still be contracted down to a single verb, instead of a verb category.</para> 
+        /// </remarks>
+        public const int MaxContract = 20;
+
+        /// <summary>
+        ///     If true, this verb category is shown in the context menu as a row of icons without any text.
+        /// </summary>
+        /// <remarks>
+        ///     For example, the 'Rotate' category simply shows two icons for rotating left and right.
+        /// </remarks>
+        public readonly bool IconsOnly;
+
+        public VerbCategory(string text, string? icon, bool contractible = false, bool iconsOnly = false)
+        {
+            Text = Loc.GetString(text);
+            Contractible = contractible;
+            Icon = icon == null ? null : new SpriteSpecifier.Texture(new ResourcePath(icon));
+            IconsOnly = iconsOnly;
+        }
+
+        public static readonly VerbCategory Debug =
+            new("verb-categories-debug", "/Textures/Interface/VerbIcons/debug.svg.192dpi.png");
+
+        public static readonly VerbCategory Eject =
+            new("verb-categories-eject", "/Textures/Interface/VerbIcons/eject.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Insert =
+            new("verb-categories-insert", "/Textures/Interface/VerbIcons/insert.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Buckle =
+            new("verb-categories-buckle", "/Textures/Interface/VerbIcons/buckle.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Unbuckle =
+            new("verb-categories-unbuckle", "/Textures/Interface/VerbIcons/unbuckle.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Open =
+            new("verb-categories-open", "/Textures/Interface/VerbIcons/open.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Close =
+            new("verb-categories-close", "/Textures/Interface/VerbIcons/close.svg.192dpi.png", true);
+
+        public static readonly VerbCategory Rotate =
+            new("verb-categories-rotate", "/Textures/Interface/VerbIcons/refresh.svg.192dpi.png", iconsOnly: true);
+
+        public static readonly VerbCategory SetTransferAmount =
+            new("verb-categories-transfer", "/Textures/Interface/VerbIcons/spill.svg.192dpi.png");
```

Comment:
this is pretty ugly.. honestly might be nice if verb categories were prototypes? i'm not sure

Reply:
I agree that  these should maybe just be prototypes. It should also reduce network bandwidth when sending verbs over the network (just the prototype id, instead of full object with text & SpriteSpecifier).

On the other hand, you would then be hardcoding a bunch of yaml prototype IDs into a lot of systems and that feels icky.